File: solutions/cpp/sum-of-multiples/1/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples {

int to(std::vector<int> const& factors, int level) {
    int sum = 0;
    for (int i = 1; i < level; ++i) {
        for (int factor : factors) {
            if (factor != 0 && i % factor == 0) {
                sum += i;
                break;  // count each i only once
            }
        }
    }
    return sum;
}

}  // namespace sum_of_multiples

File: solutions/python/sum-of-multiples/1/sum_of_multiples.py
"""Calculate the sum of unique multiples of given factors below a limit."""


def sum_of_multiples(limit: int, multiples: list[int]) -> int:
    """Return the sum of all unique multiples of the given factors below limit.

    Args:
        limit: The upper bound (exclusive).
        multiples: A list of base values (factors).

    Returns:
        The sum of all unique multiples of the factors less than limit.
    """
    unique_multiples = set()

    for factor in multiples:
        if factor == 0:
            continue
        for value in range(factor, limit, factor):
            unique_multiples.add(value)

    return sum(unique_multiples)